src/waitToLoadReact.js

```diff
@@ -48,3 +48,56 @@ export function waitToLoadReact(timeout = 5000, rootElSelector) {
         }, timeout)
     })
 }
+
+export function waitToLoadReactInIframe(timeout = 5000, iFrameElSelector, rootElSelector) {
+    if (global.isReactLoaded) {
+        return Promise.resolve('React already loaded')
+    }
+
+    const findReactRoot = () => {
+        if (iFrameElSelector && rootElSelector) {
+            return document
+                .querySelector(iFrameElSelector)
+                .contentWindow
+                .document
+                .querySelector(rootElSelector)
+        }
+
+        const walker = document.createTreeWalker(document)
+
+        while(walker.nextNode()) {
+            if (walker.currentNode.hasOwnProperty('_reactRootContainer')) {
+                return walker.currentNode
+            }
+        }
+    }
+
+    return new Promise((resolve, reject) => {
+        let timedout = false
+
+        const tryToFindApp = () => {
+            const reactRoot = findReactRoot()
+
+            if (reactRoot) {
+                global.isReactLoaded = true
+                global.rootReactElement = findReactInstance(reactRoot)
+                return resolve()
+            }
+            /* istanbul ignore next */
+            if (timedout) {
+                return
+            }
+
+            setTimeout(tryToFindApp, 200)
+        }
+
+        tryToFindApp()
+
+        /* istanbul ignore next */
+        setTimeout(() => {
+            timedout = true
+
+            reject('Timed out')
+        }, timeout)
+    })
```

Comment:
These lines of code are the same as in the `waitToLoadReact` function. Maybe we can extract these into a helper function in order to not repeat code.